desktopexporter/internal/store/queries.go

```diff
@@ -62,12 +62,49 @@ const (
 		statusCode VARCHAR, 
 		statusMessage VARCHAR)
 	`
-	SELECT_ORDERED_TRACES = `
-		SELECT traceID 
-		FROM spans
-		GROUP BY traceID
-		ORDER BY MAX(startTime) DESC
+	CREATE_ROOT_SPAN_MACRO = `
+        CREATE MACRO get_root_span_field(trace_id, field) AS (
+            SELECT field
+            FROM spans
+            WHERE traceID = trace_id
+            AND parentSpanID = ''
+            LIMIT 1
+        );
+    `
```

Comment:
I think this will execute a full table scan every time you run this macro (unless the query optimizer does some magic?). I think you probably want to use a subquery here instead like you are doing for `spanCount`

https://mode.com/sql-tutorial/sql-sub-queries
https://duckdb.org/docs/sql/expressions/subqueries.html

Reply:
Yeah, this is correct. I simplified it.